app/code/Magento/CatalogInventory/Model/StockStateProvider.php

```diff
@@ -119,13 +119,13 @@ public function checkQuoteItemQty(StockItemInterface $stockItem, $qty, $summaryQ
         $result->setItemIsQtyDecimal($stockItem->getIsQtyDecimal());
         if (!$stockItem->getIsQtyDecimal()) {
             $result->setHasQtyOptionUpdate(true);
-            $qty = (int) $qty;
+            $qty = ($qty > 1) ? (int) $qty : 1;
```

Comment:
I would simplify the condition in the following way:
```
$qty = (int) $qty ?: 1;
```

Reply:
Fixed